packages/server/graphql/public/mutations/regenerateOAuthAPIProviderSecret.ts

```diff
@@ -0,0 +1,51 @@
+import {generateOAuthClientSecret} from '../../../oauth/credentials'
+import getKysely from '../../../postgres/getKysely'
+import {getUserId, isUserOrgAdmin} from '../../../utils/authorization'
+import standardError from '../../../utils/standardError'
+import {GQLContext} from '../../graphql'
+
+interface RegenerateOAuthAPIProviderSecretInput {
+  providerId: string
+}
+
+export default async function regenerateOAuthAPIProviderSecret(
+  _root: any,
+  {input}: {input: RegenerateOAuthAPIProviderSecretInput},
+  context: GQLContext
+) {
+  const {providerId} = input
+  const {authToken, dataLoader} = context
+  const viewerId = getUserId(authToken)
+
+  const pg = getKysely()
+
+  const provider = await pg
+    .selectFrom('OAuthAPIProvider')
+    .select('organizationId')
+    .where('id', '=', providerId)
+    .executeTakeFirst()
+
+  if (!provider) {
+    throw new Error('Provider not found')
+  }
+
+  if (!(await isUserOrgAdmin(viewerId, provider.organizationId, dataLoader))) {
+    return standardError(new Error('Not organization lead'), {
+      userId: viewerId
+    })
+  }
+
+  const newSecret = generateOAuthClientSecret()
+
+  const updatedProvider = await pg
+    .updateTable('OAuthAPIProvider')
+    .set({
+      clientSecret: newSecret,
+      updatedAt: new Date()
```

Comment:
this mutation ha the same fixups as commented in updateOAuthAPIProvider.
types should come from the auto generated gql types, updatedAt should be set by the DB via trigger, use `GraphQLError` instead of a Success object, etc.